### PR TITLE
[SPARK-27757][BUILD] Bump Jackson to 2.9.9

### PR DESCRIPTION
**Pull request by Fokko**
## What changes were proposed in this pull request?

This fixes CVE-2019-12086 on Databind: https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.9.9

## How was this patch tested?

Existing tests

### CODE REVIEW

**Comment by srowen** [comment]
You'll have to run `dev/test-dependencies.sh --replace-manifest` too.

File: pom.xml [hunk 1]
```diff
@@ -166,7 +166,7 @@
     <!-- for now, not running scalafmt as part of default verify pipeline -->
     <scalafmt.skip>true</scalafmt.skip>
     <codehaus.jackson.version>1.9.13</codehaus.jackson.version>
-    <fasterxml.jackson.version>2.9.8</fasterxml.jackson.version>
+    <fasterxml.jackson.version>2.9.9</fasterxml.jackson.version>
```
Comment (dongjoon-hyun) [inline pom.xml]: When this is uploaded to the Maven Central, please update the dependency with @srowen 's comment.

**Comment by Fokko** [comment]
```
MacBook-Pro-van-Fokko:spark fokkodriesprong$ dev/test-dependencies.sh --replace-manifest
Using `mvn` from path: /usr/local/bin/mvn
Using `mvn` from path: /usr/local/bin/mvn
Performing Maven install for hadoop-2.7
Using `mvn` from path: /usr/local/bin/mvn
[ERROR] Failed to execute goal on project spark-core_2.12: Could not resolve dependencies for project org.apache.spark:spark-core_2.12:jar:spark-672604: Could not find artifact com.fasterxml.jackson.module:jackson-module-scala_2.12:jar:2.9.9 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :spark-core_2.12
Using `mvn` from path: /usr/local/bin/mvn
```

Seems like `org.apache.spark:spark-core_2.12:jar` is missing for 2.9.9: https://mvnrepository.com/artifact/com.fasterxml.jackson.module/jackson-module-scala

There is a pending issue here: https://github.com/FasterXML/jackson-module-scala/issues/412

**Comment by dongjoon-hyun** [comment]
? @Fokko . Your error message is saying `jackson-module-scala_2.12:jar:2.9.9` is not at Maven Central. Please see the part `Could not find artifact com.fasterxml.jackson.module:jackson-module-scala_2.12:jar:2.9.9 in central (https://repo.maven.apache.org/maven2)`. That is the root cause of building and the reason I recommend you to check Maven Central for 2.9.9.

**Comment by dongjoon-hyun** [comment]
For now, I'll close this PR to reduce the visibility, @Fokko . Feel free to click `reopen` on this PR when 2.9.9 is ready on Maven Central.

**Comment by Fokko** [comment]
@dongjoon-hyun The artifact has been published. However I don't see the `reopen` button, I think there is something off.

**Comment by srowen** [comment]
Hm yeah I can't reopen either. That's new, and we might need to investigate. Could you just open a new one in the meantime? or try pushing a new commit here?

**Comment by Fokko** [comment]
No problem, I already created a new one under https://github.com/apache/spark/pull/24646

**Comment by dongjoon-hyun** [comment]
Oh.. It's weird. Sorry for the inconvenience, @Fokko . :(

**Pull request closed without merging**